bin/create-files.js

```diff
@@ -1,49 +1,113 @@
-#!/usr/bin/env node
+#!/usr/bin/env node
 
-const fs = require("fs")
-const process = require("process")
+const fs = require("fs");
+const process = require("process");
 
-const readAndWriteFiles = (projectDir) => {
+if (process.argv[2] === undefined) {
+  const readAndWriteFiles = (projectDir) => {
+    //console.log(projectDir);
     fs.readFile(__dirname + "/index.html", (err, data) => {
+      if (err) {
+        throw err;
+      }
+      fs.writeFile(projectDir + "/index.html", data, (err) => {
         if (err) {
-            throw err
+          throw err;
         }
-        fs.writeFile(projectDir + "/index.html", data, (err) => {
-            if (err) {
-                throw err
-            }
-            console.log("Created index.html")
-        })
-    })
+        console.log("Created index.html");
+      });
+    });
 
     // create an empty file
     fs.open(projectDir + "/script.js", "w", (err, file) => {
-        if (err) {
-            throw err
-        }
+      if (err) {
+        throw err;
+      }
 
-        console.log("Created script.js")
-    })
+      console.log("Created script.js");
+    });
 
     fs.open(projectDir + "/style.css", "w", (err, file) => {
-        if (err) {
-            throw err
-        }
+      if (err) {
+        throw err;
+      }
 
-        console.log("Created style.css")
-    })
-}
+      console.log("Created style.css");
+    });
+  };
 
-const dirToCreate = process.argv[2]
-let projectDir = process.cwd()
-if (!dirToCreate) {
-    readAndWriteFiles(projectDir)
-} else {
-    projectDir = projectDir + "/" + dirToCreate
+  const dirToCreate = process.argv[2];
+  let projectDir = process.cwd();
+  if (!dirToCreate) {
+    readAndWriteFiles(projectDir);
+  } else {
+    projectDir = projectDir + "/" + dirToCreate;
     fs.mkdir(projectDir, (err) => {
+      if (err) {
+        throw err;
+      }
+      readAndWriteFiles(projectDir);
+    });
+  }
+} else {
+  const readAndWriteFiles = (projectDir) => {

```

Comment:
Instead of redefining readAndWriteFiles again in else condition,what you should do is accept file names as arguments or in the function itself ,extract them from process.argv

Reply:
sir so the else part only comes when we have to provide our customed named files other wise we have to use our default named file, is it?